dvc/repo/ls.py

```diff
@@ -82,3 +130,26 @@ def _get_target_path_info(repo, target=None):
     if not target:
         return PathInfo(repo.root_dir)
     return PathInfo(repo.root_dir, target)
+
+
+def _get_fs_node(path_info, out=None):
+    from dvc.compat import fspath
+
+    if out:
+        isdir = out.is_dir_checksum if out.checksum else False
```

Comment:
If there is not checksum it might still be dir, I would write a note here about this edge case. Now the code looks quirky for no apparent reason.

Reply:
As I got if `out.checksum` is missed than `out.is_dir_checksum` fails
Do you have thoughts how can it be improved?

Reply:
Yeah, it fails, we don't really no whether that is a dir. We may just say about this limitation in a comment.

Reply:
> we don't really know whether that is a dir

In what case the out can be dir but doesn't have `checksum`?

Reply:
Yes.

Reply:
@Suor it was an open question - could you please provide the case when it can happen?

Reply:
When a user edits a `.dvc` file and deletes the output checksum there. This is not as weird as it might sound, people may want to do that to recalc the checksum from the actual dir they have.

Reply:
Ok, I can imagine that user don't see folder when has been working on dvc file and doesn't understand what's going on, but it's really specific case. I can handle it in the https://github.com/iterative/dvc/issues/3394. What do you think @efiop?